### PR TITLE
Improve tap detection

### PR DESCRIPTION
**Pull request by Stermere**
Improves the algorithm by checking for a moment of low acceleration before allowing another tap to be registered. It also fixes a logic bug and gives a config option to allow IMUs with a faulty accelerometer but a good gyro to not break this feature.

edit: also fixes a bug with Vive emulation not saving and allows for up to ten taps to be the quantity before a reset

### CODE REVIEW

File: server/src/main/java/dev/slimevr/config/TapDetectionConfig.java [hunk 1]
```diff
@@ -72,8 +73,8 @@ public int getQuickResetTaps() {
 
 	// clamp to 2-3 to prevent errors
 	public void setQuickResetTaps(int quickResetTaps) {
-		if (quickResetTaps > 3 || quickResetTaps < 2)
-			FastMath.clamp(quickResetTaps, 2, 3);
+		if (quickResetTaps > 10 || quickResetTaps < 2)
+			FastMath.clamp(quickResetTaps, 2, 10);
```
Comment (Eirenliel) [inline server/src/main/java/dev/slimevr/config/TapDetectionConfig.java:77]:
```suggestion
		this.quickResetTaps = FastMath.clamp(quickResetTaps, 2, 10);
```

File: server/src/main/java/dev/slimevr/config/TapDetectionConfig.java [hunk 2]
```diff
@@ -72,8 +73,8 @@ public int getQuickResetTaps() {
 
 	// clamp to 2-3 to prevent errors
 	public void setQuickResetTaps(int quickResetTaps) {
-		if (quickResetTaps > 3 || quickResetTaps < 2)
-			FastMath.clamp(quickResetTaps, 2, 3);
+		if (quickResetTaps > 10 || quickResetTaps < 2)
```
Comment (Eirenliel) [inline server/src/main/java/dev/slimevr/config/TapDetectionConfig.java:76]:
```suggestion
```

File: server/src/main/java/dev/slimevr/config/TapDetectionConfig.java [hunk 3]
```diff
@@ -82,8 +83,8 @@ public int getResetTaps() {
 	}
 
 	public void setResetTaps(int resetTaps) {
-		if (resetTaps > 3 || resetTaps < 2)
-			FastMath.clamp(resetTaps, 2, 3);
+		if (resetTaps > 10 || resetTaps < 2)
```
Comment (Eirenliel) [inline server/src/main/java/dev/slimevr/config/TapDetectionConfig.java:86]:
```suggestion
```

File: server/src/main/java/dev/slimevr/config/TapDetectionConfig.java [hunk 4]
```diff
@@ -82,8 +83,8 @@ public int getResetTaps() {
...
+			FastMath.clamp(resetTaps, 2, 10);
```
Comment (Eirenliel) [inline server/src/main/java/dev/slimevr/config/TapDetectionConfig.java:87]:
```suggestion
		this.resetTaps = FastMath.clamp(resetTaps, 2, 10);
```

File: server/src/main/java/dev/slimevr/config/TapDetectionConfig.java [hunk 5]
```diff
@@ -92,8 +93,16 @@ public int getMountingResetTaps() {
 	}
 
 	public void setMountingResetTaps(int mountingResetTaps) {
-		if (mountingResetTaps > 3 || mountingResetTaps < 2)
-			FastMath.clamp(mountingResetTaps, 2, 3);
+		if (mountingResetTaps > 10 || mountingResetTaps < 2)
```
Comment (Eirenliel) [inline server/src/main/java/dev/slimevr/config/TapDetectionConfig.java:96]:
```suggestion
```

File: server/src/main/java/dev/slimevr/config/TapDetectionConfig.java [hunk 6]
```diff
@@ -92,8 +93,16 @@ public int getMountingResetTaps() {
...
+			FastMath.clamp(mountingResetTaps, 2, 10);
```
Comment (Eirenliel) [inline server/src/main/java/dev/slimevr/config/TapDetectionConfig.java:97]:
```suggestion
		this.mountingResetTaps = FastMath.clamp(mountingResetTaps, 2, 10);
```

**Review by Eirenliel — CHANGES_REQUESTED** [review]

**Review by Eirenliel — APPROVED** [review]

**Comment by Eirenliel** [comment]
nya_a conflict plz resolve 🥺

File: gui/src/components/settings/pages/GeneralSettings.tsx [hunk 7]
```diff
@@ -539,8 +539,10 @@ export function GeneralSettings() {
                   variant="toggle"
                   outlined
                   control={control}
-                  name="toggles.viveEmulation"
-                  label={t('settings-general-fk_settings-vive_emulation-label')}
+                  name="toggles,viveEmulation"
```
Comment (Eirenliel) [inline gui/src/components/settings/pages/GeneralSettings.tsx:542]: I am very sure that is a wrong one, lol

**Review by Eirenliel — CHANGES_REQUESTED** [review]

File: gui/src/components/settings/pages/GeneralSettings.tsx (hunk 7 above)
Comment (Stermere) [reply gui/src/components/settings/pages/GeneralSettings.tsx:542]: did that recently get changed to a comma? I'll try tho

**Comment by Stermere** [comment]
Don't merge it yet its still acting strange

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**